functests/utils/pods/pods.go

```diff
@@ -40,6 +44,80 @@ func GetBusybox() *corev1.Pod {
 	}
 }
 
+func GetDefinition(namespace string) *corev1.Pod {
+	podObject := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "testpod-",
+			Namespace:    namespace},
+		Spec: corev1.PodSpec{
+			TerminationGracePeriodSeconds: pointer.Int64Ptr(0),
+			Containers: []corev1.Container{{Name: "test",
+				Image:   "quay.io/schseba/utility-container:latest",
+				Command: []string{"/bin/bash", "-c", "sleep INF"}}}}}
+
+	return podObject
+}
+
+// RedefineWithCommand updates the pod defintion with a different command
+func RedefineWithCommand(pod *corev1.Pod, command []string, args []string) *corev1.Pod {
+	pod.Spec.Containers[0].Command = command
+	pod.Spec.Containers[0].Args = args
+	return pod
+}
+
+// RedefineWithRestartPolicy updates the pod defintion with a restart policy
+func RedefineWithRestartPolicy(pod *corev1.Pod, restartPolicy corev1.RestartPolicy) *corev1.Pod {
+	pod.Spec.RestartPolicy = restartPolicy
+	return pod
+}
+
+func DefineWithHostNetwork(podObject *corev1.Pod, nodeName string) *corev1.Pod {
+	podObject.Spec.NodeSelector = map[string]string{
+		"kubernetes.io/hostname": nodeName,
+	}
+
+	return podObject
+}
+
+func RedefineWithHugePage(namespace, nodeName, writeSize string) *corev1.Pod {
```

Comment:
This can be "DefineWithHugePages" as it's not accepting a pod definition to redefine, or you can change it to accept a pod definition instead of adding a new one.